ostree-ext/src/container/deploy.rs

```diff
@@ -122,6 +136,24 @@ pub async fn deploy(
             flags,
             cancellable,
         )?;
+
+        // We end up re-executing ourselves as a subprocess because
+        // otherwise right now we end up with a circular dependency between
+        // crates. We need an option to skip though so when the *main*
+        // bootc install code calls this API, we don't do this as it
+        // will have already been handled.
+        if !options.skip_completion {
```

Comment:
Does this mean that all future users of `ostree container image deploy`  (where options.skip_completion` is `false`) will have bootc completion running for them? 

What about users doing it on non-bootc images? For missing InstallConfiguration (kargs) I see we use Option::into_iter() which will make it a no-op

Similarly for bound images, it will just be empty

So it should work in theory, but did you test this in practice? that the command is happy with completely non-bootc images?

Reply:
I'm basically afraid that this PR will introduce a regression / unexpected errors in users of `ostree container image deploy` in non-bootc images

Reply:
I didn't explicitly test that yet, and we should. However note that none of this code  will run *in practice* until the moment when we take over from the version in rpm-ostree.

Reply:
But yes basically the heart of this right now is looking for files (kargs, bound image symlinks) and acting on them if present, and if they don't it's a no-op.